### PR TITLE
Adding storage factories for owned objects.

### PR DESCRIPTION
**Pull request by dhermes**
This includes
- `Bucket<--Client` (bucket owned by client)
- `Batch<--Client`
- `Blob<--Bucket`

I refrained from implementing factories for
- `BucketACL<--Bucket`
- `ObjectACL<--Blob`

### CODE REVIEW

File: gcloud/storage/test_bucket.py [hunk 1]
```diff
@@ -109,6 +109,19 @@ def test_ctor(self):
...
+        BUCKET_NAME = 'BUCKET_NAME'
+        BLOB_NAME = 'BLOB_NAME'
+
+        bucket = self._makeOne(name=BUCKET_NAME)
+        blob = bucket.blob(BLOB_NAME)
+        self.assertTrue(isinstance(blob, Blob))
+        self.assertTrue(blob.bucket is bucket)
+        self.assertTrue(blob.client is bucket.client)
+        self.assertEqual(blob.name, BLOB_NAME)
+
```
Comment (tseaver) [inline gcloud/storage/test_bucket.py]: Needs a test for explicit `chunk_size`.

Comment (dhermes) [inline gcloud/storage/test_bucket.py]: Good call. I meant to do it, but then forgot when implementing :smile:

**Comment by dhermes** [comment]
Rebased on top of #1002, need to re-address some of the places in docs mentioned by @jgeewax

**Comment by dhermes** [comment]
@tseaver Folded the `chunk_size` check into the original commit (and also changed a lot of `storage.Blob` instances to `bucket.blob` in the docs).

**Comment by tseaver** [comment]
LGTM

**Pull request merged**